File: Caper/Caper/Mapping.cpp
#include "Mapping.h"

Mapping::Mapping(string & aName, int & aIndex, Sequence* aSequence, Orientation aStrand)
{
  Name = aName;
	Index = aIndex;
	mSequence = aSequence;
  Strand = aStrand;
}

Mapping::Mapping( Mapping & aMapping )
{
  Name = aMapping.Name;
  Index = aMapping.Index;
  mSequence = new Sequence( *(aMapping.mSequence) );
  Strand = aMapping.Strand;
}

Mapping::~Mapping(void)
{
  delete mSequence;
}

File: Caper/CaperSharp/Mapping.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CaperSharp
{
  internal class Mapping
  {
    public int Index { get; private set; }
    public Sequence Sequence { get; private set; }

    public Mapping( int aIndex, Sequence aSequence )
    {
      Index = aIndex;
      Sequence = aSequence;
    }
  }
}
